paymentsheet/src/main/java/com/stripe/android/customersheet/injection/CustomerSheetViewModelModule.kt

```diff
@@ -177,6 +180,15 @@ internal interface CustomerSheetViewModelModule {
             return DefaultEditPaymentMethodViewInteractor.Factory
         }
 
+        @Provides
+        fun providesErrorReporter(
+            analyticsRequestFactory: AnalyticsRequestFactory,
+            analyticsRequestExecutor: DefaultAnalyticsRequestExecutor,
+        ): ErrorReporter = RealErrorReporter(
```

Comment:
`CustomerSheetViewModelModule` provides a base `AnalyticsRequestFactory` and not the `PaymentAnalyticsRequestFactory`. Should we always defer to `PaymentAnalyticsRequestFactory`?

Reply:
Actually I think we should defer to `PaymentAnalyticsRequestFactory`. There has been an issue with `PaymentSheet` and `FlowController` not properly attributing its product usage with a lot of its request and it looks like it's in relation to the usage of the base `AnalyticsRequestFactory` which doesn't send product usage by default.